[MOB-4963] Update Web SDK README (#175)
Iterable/iterable-web-sdk: README.md

Below are the methods this SDK exposes. See [Iterable's API Docs](https://api.iterable.com/api/docs) for information on what data to pass and what payload to receive from the HTTP requests.

| Method Name                                            | Description                                                                                                                         |
| ------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------- |

> | [`getInAppMessages`](#getInAppMessages)                | Either return in-app messages as a Promise or automatically paint them to the DOM if the second argument is passed `DisplayOptions` |

Reviewer: @pauljung14 Do you think we could just say "display them" rather than "paint them to the DOM?"

Reviewer: @pauljung14 Also, this method _always_ fetches messages, right? And then either displays or doesn't display them? I think the way this is phrased, it reads more like "either fetch or display"

Author: Lemme rewrite that to make it clearer

> | [`initialize`](#initialize)                            | Method for identifying users and setting a JWT                                                                                      |

Reviewer: @pauljung14 Isn't `setEmail` for identifying the user?

Author: That's correct. So the flow goes like this: `setEmail` is part of the return signature for `initialize`. Once you initialize with an auth token an a callback to generate your JWT, then you can call setEmail to identify your user.

Reviewer: Got it. Wondering if it's confusing to describe this method as being used to identify the user, when it's really a follow-on call?

Author: Updating

| [`track`](#track)                                      | Track custom events                                                                                                                 |
| [`trackInAppClick`](#trackInAppClick) :rotating_light: | Track when a user clicks on a button or link within a message                                                                       |
| [`trackInAppClose`](#trackInAppClose)                  | Track when an in-app message is closed                                                                                              |

> | [`trackInAppConsume`](#trackInAppConsume)              | Track when a message has been consumed. Deletes the in-app message from the server so it won't be returned anymore                  |

Reviewer: @pauljung14 Can we give any guidance as to when this should be called? Right after displaying the message?

> | [`trackInAppConsume`](#trackInAppConsume)              | Track when a message has been consumed. Deletes the in-app message from the server so it won't be returned anymore                  |

Reviewer: ```suggestion
| [`trackInAppConsume`](#trackInAppConsume)              | Track when a message has been consumed and delete the in-app message from the server so it won't be returned anymore.                  |
```

| [`trackInAppDelivery`](#trackInAppDelivery)            | Track when a message has been delivered to a user's device                                                                          |

> | [`trackInAppOpen`](#trackInAppOpen)                    | Track when a message is opened and marks it as read                                                                                 |

Reviewer: ```suggestion
| [`trackInAppOpen`](#trackInAppOpen)                    | Track when a message is opened and mark it as read                                                                                 |
```

| [`trackPurchase`](#trackPurchase)                      | Track purchase events                                                                                                               |

> | [`updateCart`](#updateCart)                            | Update _shoppingCartItems_ field on user profile                                                                                    |

Reviewer: ```suggestion
| [`updateCart`](#updateCart)                            | Update `shoppingCartItems` field on user profile                                                                                    |
```

> | [`updateSubscriptions`](#updateSubscriptions)          | Updates user's subscriptions                                                                                                        |

Reviewer: ```suggestion
| [`updateSubscriptions`](#updateSubscriptions)          | Update the user's subscriptions.                                                                                                        |
```

> | [`updateCart`](#updateCart)                            | Update _shoppingCartItems_ field on user profile                                                                                                                       |

Reviewer: ```suggestion
| [`updateCart`](#updateCart)                            | Update `shoppingCartItems` field on user profile                                                                                                                       |
```

| [`updateUser`](#updateUser)                            | Change data on a user's profile or create a user if none exists                                                                     |

> | [`updateUserEmail`](#updateUserEmail)                  | Change a user's email and reauthenticate user with the new email address (in other words, we will call `setEmail` for you)          |

Reviewer: ```suggestion
| [`updateUserEmail`](#updateUserEmail)                  | Change a user's email and reauthenticate user with the new email address (in other words, the SDK calls `setEmail` for the new email)          |
```

> :rotating_light: Due to a limitation in Safari browsers, web in-app messages displayed in Safari will not be able to fire `trackInAppClick` events automatically on all links in the template. This will impact analytics for users on Safari.

Reviewer: ```suggestion
:rotating_light: Due to a limitation in Safari browsers, web in-app messages displayed in Safari can't automatically fire `trackInAppClick` events when a link has been clicked. This will impact analytics for Safari users.
```

API [(see required API payload here)](https://api.iterable.com/api/docs#In-app_getMessages):

```ts
getInAppMessages: (payload: InAppMessagesRequestParams, options?: { display: 'deferred' | 'immediate' }) => Promise<TrackConsumeData> | PaintInAppMessageData
```

> :rotating_light: Notice: As of v1.0.0 we have deprecated support for `showMessagesAutomatically?: boolean`. If needed, please update your getInAppMessages requests to use `options: { display: 'deferred' | 'immediate' }` instead.

Reviewer: ```suggestion
:rotating_light: Notice: v1.0.0 of this SDK deprecates support for `showMessagesAutomatically?: boolean`. If needed, please update your getInAppMessages requests to use `options: { display: 'deferred' | 'immediate' }` instead.
```

SDK Specific Options:

Along with the API parameters, you can pass these options to the SDK method to have in-app messages behave differently.

| Property Name                | Description                                                                                                                                                                                                                                                                                | Value                                                             | Default     |
| ---------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------- | ----------- |

> | animationDuration            | How much time (in MS) for messages to animate in and out                                                                                                                                                                                                                                   | `number`                                                          | `400`       |

Reviewer: ```suggestion
| animationDuration            | How much time (in ms) for messages to animate in and out                                                                                                                                                                                                                                   | `number`                                                          | `400`       |
```

> | bottomOffset                 | How much space (px or %) to create between the bottom of the screen and messages. Not applicable for center, top, or full-screen messages                                                                                                                                                  | `string`                                                          | `undefined` |

Reviewer: ```suggestion
| bottomOffset                 | How much space (px or %) to create between the bottom of the screen and a message. Not applicable for center, top, or full-screen messages.                                                                                                                                                  | `string`                                                          | `undefined` |
```

> | displayInterval              | How much time (in MS) to wait before showing next in-app message after closing the currently opened one                                                                                                                                                                                    | `number`                                                          | `30000`     |

Reviewer: ```suggestion
| displayInterval              | An amount of time (in MS) to wait before showing next in-app message after closing the currently opened one                                                                                                                                                                                    | `number`                                                          | `30000`     |
```

| handleLinks                  | How to open links. If `undefined`, use browser-default behavior. `open-all-new-tab` opens all in new tab, `open-all-same-tab` opens all in same tab, `external-new-tab` opens only off-site links in new tab, otherwise same tab. Overrides the target attribute defined on link elements. | `'open-all-new-tab' \| 'open-all-same-tab' \| 'external-new-tab'` | `undefined` |

> | onOpenScreenReaderMessage    | What text do you want the screen reader to announce when opening in-app messages                                                                                                                                                                                                           | `string`                                                          | `undefined` |

Reviewer: ```suggestion
| onOpenScreenReaderMessage    | The text a screen reader should read when opening in-app messages.                                                                                                                                                                                                           | `string`                                                          | `undefined` |
```

> | onOpenNodeToTakeFocus        | What DOM element do you want to take keyboard focus when the in-app message opens. (Will open the first interact-able element if not specified). Any query selector is valid.                                                                                                              | `string`                                                          | `undefined` |

Reviewer: ```suggestion
| onOpenNodeToTakeFocus        | The DOM element that should receive keyboard focus when the in-app message opens. Any query selector is valid. If not specified, the first interactive element receives focus.                                                                                                               | `string`                                                          | `undefined` |
```

> | rightOffset                  | How much space (px or %) to create between the right of the screen and messages. Not applicable for center or full-screen messages                                                                                                                                                         | `string`                                                          | `undefined` |

Reviewer: ```suggestion
| rightOffset                  | How much space (px or %) to create between the right of the screen and messages. Not applicable for center or full-screen messages.                                                                                                                                                         | `string`                                                          | `undefined` |
```

> | topOffset                    | How much space (px or %) to create between the top of the screen and messages. Not applicable for center, bottom, or full-screen messages                                                                                                                                                  | `string`                                                          | `undefined` |

Reviewer: ```suggestion
| topOffset                    | How much space (px or %) to create between the top of the screen and the message. Not applicable for center, bottom, or full-screen messages.                                                                                                                                            | `string`                                                          | `undefined` |
```

> | closeButton :rotating_light: | Properties to show a custom close button on each in-app message                                                                                                                                                                                                                            | `CloseButtonOptions` (see below)                                  | `undefined` |

Reviewer: ```suggestion
| closeButton :rotating_light: | Properties that define a custom close button to display on the message.                                                                                                                                                                                                                            | `CloseButtonOptions` (see below)                                  | `undefined` |
```

> :rotating_light: Due to a limitation in Safari browsers, web in-app messages displayed in Safari will not be able to add an auto-generated close button and automatically fire `trackInAppClose` events with it for you.

Reviewer: @pauljung14 Can safari create the button, but not fire the events? Or neither? Can we clarify this?

Author: Got it, will clarify

Close Button Options:

| Property Name              | Description                                                                           | Value                      | Default       |
| -------------------------- | ------------------------------------------------------------------------------------- | -------------------------- | ------------- |

> | color                      | What color the button is (does not affect custom icons)                               | `string`                   | `undefined`   |

Reviewer: ```suggestion
| color                      | The button's color (does not affect custom icons)                               | `string`                   | `undefined`   |
```

> | iconPath                   | Custom pathname to an image or SVG you want to show instead of the default "X"        | `string`                   | `undefined`   |

Reviewer: ```suggestion
| iconPath                   | Custom pathname to an image or SVG to use (instead of the default "X")        | `string`                   | `undefined`   |
```

> | position                   | Where to appear relative to the in-app message                                        | `'top-right \| 'top-left'` | `'top-right'` |

Reviewer: ```suggestion
| position                   | Where the button should display on an in-app message                                        | `'top-right \| 'top-left'` | `'top-right'` |
```

> | isRequiredToDismissMessage | If `true`, prevent user from dismissing in-app message by clicking outside of message | `boolean`                  | `undefined`   |

Reviewer: ```suggestion
| isRequiredToDismissMessage | If `true`, users cannot dismiss in-app messages by clicking outside of them. | `boolean`                  | `undefined`   |
```
